src/App.tsx

```diff
@@ -1,32 +1,69 @@
 import { Header } from './Header/Header.tsx';
-import { Card, GenreSelect, SearchForm } from './components';
+import { ApplicationName, Card, Dialog, SearchForm, SearchIcon, SortBy } from './components';
+import { AddMovieButton, MovieDetails, MovieModel,GenreSelect, MoviesList, MovieForm} from './Movies';
+import { Footer } from './Footer/Footer.tsx';
 
 function App(): ReactElement {
-	const [count, setCount] = useState(0);
 	const [searchTerm, setSearchTerm] = useState('');
+	const [selectedMovie, setSelectedMovie] = useState<MovieModel | null>(null);
 	const [selectedGenre, setGenre] = useState<string | null>('All');
-	const [genres] = useState(['All', 'Documentary', 'Comedy', 'Horror', 'Crime']);
+	const [selectedSortOption, setSortOption] = useState<string | null>('Release Date');
+	const genres = ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'];
+	const sortByOptions = ['Release Date', 'Title'];
+	const [isDialogOpen, setIsDialogOpen] = useState(false);
 
-	return (
-		<>
-			<Header>
-				<SearchForm searchTerm={searchTerm} onSearch={setSearchTerm}></SearchForm>
-			</Header>
-			<main>
+	const movieDetails: MovieModel = {
+		id: '12',
+		image: PosterImg, name: 'SpiderMan', releaseYear: '2004',
+		genres: ['Fight', 'Fantastic'],
+		movieUrl: 'url',
+		rating: 8.9,
+		duration: '2h 34min',
+		description: `Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). 
+		Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. 
+		Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. 
+		The lives of these seemingly unrelated people are woven together comprising of a series of funny, 
+		bizarre and uncalled-for incidents.—Soumitra`
+	};
+	return (<>
+		{
+			selectedMovie ? (
 				<Card>
-					<GenreSelect selectedGenre={selectedGenre} genres={genres} onSelect={setGenre}></GenreSelect>
-					<Counter count={count}
-						onIncrease={() => setCount((prev) => prev + 1)}
-						onDecrease={() => setCount((prev) => prev === 0 ? 0 : prev - 1)}></Counter>
+					<div className="flex-between">
+						<ApplicationName /> <SearchIcon onClick={() => setSelectedMovie(null)}></SearchIcon>
+					</div>
+					<MovieDetails movie={movieDetails} />
 				</Card>
-			</main>
+			) : (
+				<Header>
+					<div className="search-app-header flex-between">
+						<ApplicationName /> <AddMovieButton onClick={() => setIsDialogOpen(true)} />
+					</div>
+					<div className="movie-search">
+						<h2 className="search-field-title">Find your movie</h2>
+						<SearchForm searchTerm={searchTerm} onSearch={setSearchTerm} />
+					</div>
+				</Header>
+			)}
 
-		</>
-	);
+		<main>
+			<Dialog width={'60rem'} height={'40rem'}  title={'Add new Movie'} isOpen={isDialogOpen} onClose={() => setIsDialogOpen(false)}>
```

Comment:
What is the best practice to show a dialog? Currently, I manage the state by passing a boolean flag into Dialog. 

But I have seen a pattern where they manage like this`{isOpen && Dialog}` 

- In the first approach, we reduce the boilerplate a little bit but keep the component alien
- in the second approach we have a little bit boilerplate but component is destroyed

Reply:
Both approaches acceptable. Depends on requirements. I preffere to fully unmout component from DOM (`{isOpen && <Dialog />}`). But in this case you can't keep data entered by user and might be issue with CSS animation if you have such. So choose what suits to your needs

Reply:
Okay, thank you